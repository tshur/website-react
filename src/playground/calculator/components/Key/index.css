:local .key {
  font-family: 'Courier New', Courier, monospace;
  font-size: 24px;
}

:local .wide {
  width: 135px !important;
}