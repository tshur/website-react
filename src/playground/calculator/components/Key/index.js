import React from 'react';
import styles from './index.css';
import styled from 'styled-components';

const Button = styled.button`
  background: #eee;
  border: 1px solid black;
  height: 60px;
  width: 60px;
  cursor: pointer;
  &:focus {
    outline: 0;
  }
  &:hover {
    background: #ddd;
  }
`;

const Key = ({ symbol, onClick }) =>
  <div className={styles.key}>
    <Button
      className={symbol === '0' ? styles.wide : ''}
      onClick={onClick}
    >
      {symbol}
    </Button>
  </div>

export default Key;
