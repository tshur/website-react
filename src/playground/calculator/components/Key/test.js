import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Key from '.';

describe('Key', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Key />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const wrapper = shallow(
      <Key />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
