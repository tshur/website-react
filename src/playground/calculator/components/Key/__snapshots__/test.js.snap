// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key has a valid snapshot 1`] = `
<div>
  <styled.button
    className=""
  />
</div>
`;
